Makefile.js

```diff
@@ -390,10 +390,15 @@ target.all = function() {
 
 target.lint = function() {
     var errors = 0,
-        lastReturn;
+        lastReturn,
+        MAKEFILE_CACHE,
+        JS_CACHE,
+        TEST_CACHE;
+
+    MAKEFILE_CACHE = process.env.TRAVIS ? " " : " --cache --cache-file .cache/makefile_cache ";
```

Comment:
cache is kept only between executions with the same set of files, this is to make sure the cache won't remember files that were deleted or renamed.

Since the linting process in the makefile happens in 3 fases, we need 3 different caches for them